lib/streamlit/navigation/page.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import types
+from pathlib import Path
+from typing import Callable
+
+from streamlit.errors import StreamlitAPIException
+from streamlit.runtime.metrics_util import gather_metrics
+from streamlit.runtime.scriptrunner.script_run_context import get_script_run_ctx
+from streamlit.source_util import page_icon_and_name
+from streamlit.string_util import validate_icon_or_emoji
+from streamlit.util import calc_md5
+
+
+@gather_metrics("Page")
+def Page(
+    page: str | Path | Callable[[], None],
+    *,
+    title: str | None = None,
+    icon: str | None = None,
+    default: bool = False,
+):
+    """Configure a page in `st.navigation` in a multipage app.
+
+    The Page object is passed to `st.navigation` and returned when the user
+    navigates to that page. Call `Page.run()` on the returned Page in your
+    main script to execute the page code.
+
+    Page code can be specified by file path of the page (relative to the main
+    script) or by passing a Callable such as a function.
+
+    Parameters
+    ----------
+
+    page: str or Path or callable
+        The path to the script file or a callable that defines the page.
+
+    title: str or None
+        The title of the page. If None, the title will be inferred from the
+        page path or callable name.
+
+    icon: str or None
+        The icon of the page. If None, the icon will be inferred from the
+        page path.
```

Comment:
nit: not sure how we handle it in different parts of the library, but this is the unicode representation of an icon as string, right? if so, it might be worth to clarify it here